src/uk/ac/cam/cl/retailcategorymapper/marshalling/XMLParser.java

```diff
@@ -0,0 +1,221 @@
+package uk.ac.cam.cl.retailcategorymapper.marshalling;
+
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.w3c.dom.Node;
+import org.w3c.dom.Element;
+import org.xml.sax.SAXException;
+
+import uk.ac.cam.cl.retailcategorymapper.entities.Product;
+import uk.ac.cam.cl.retailcategorymapper.entities.ProductBuilder;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+
+public class XMLParser {
+
+	/*
+	 * Authored by Charlie
+	 * 
+	 * this class provides a method for parsing an XML file into our internal
+	 * product description class it takes a file path to an XML file and also a
+	 * file path to a file describing the format of the XML file and parses it
+	 * appropriately
+	 * 
+	 * The XML format descriptor takes the form of lines with a single ':'
+	 * separating what we call the field internally from what it is called in the
+	 * XML file ie "our name:XML field" . so an example file is
+	 *
+	 * XMLdescritor.txt
+	 * 
+	 * product:product
+   * price:productPrice
+   * id:productSku
+   * name:productName
+   * description:productDescription
+   * category:none
+   * attributes:none
+	 * 
+	 * all 7 of these fields must be provided - none after the colon indicated 
+	 * we don't what that field processed
+	 * 
+	 * Most of the code was copied from some web-site so I don't really understand
+	 * what it is doing but it seems to work.
```

Comment:
well it was from a tutorial website saying that the founder "love java and open source" so I presume its ok - no formal licensing though